js/app/core/objectProperty.filter.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+((app) => {
+	const objectPropertyFilter = () => {
+		const filterObjects = (objectList, selectedItems, targetProperty) => {
+			if (!selectedItems || !selectedItems.length) {
+				return objectList;
+			}
+			const filtered = [];
+
+			objectList.forEach((obj) => {
+				let numberOfMatches = 0;
+				for (let i = 0, len = selectedItems.length; i < len; i += 1) {
+					const category = selectedItems[i];
+
+					if (obj[targetProperty].indexOf(category) > -1) {
```

Comment:
You might want to make sure the property exists first with obj.hasOwnProperty().

Reply:
done